src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,207 @@
+                e.printStackTrace();
+            }
+        }
+
+        return true;
+    }
 
     @Override
     public void put(K key, V value) {
 
-    }
+        // Resizing
+        int threshold = (int) (loadFactor * getTotalSize());
+        int currentSize = this.getSize();
+        if (currentSize + 1 > threshold){
+            this.resize();
+        }
+
+        // Getting of element index
+        int h = this.hash(key, value);
+        int index = h % getTotalSize();
 
+        // If the key exists, we change the value
+        for (int i = 0; i < table.length; i++){
+            if (table[i] instanceof Node) {
+                if(table[i].key instanceof Object && compareObjects(table[i].key, key)){
+                    table[i].value = value;
+                    return;
+                }
+                else if(table[i].key == key){
+                    table[i].value = value;
+                    return;
+                }
+            }
+        }
+
+        // If the key doesn't exist, we create a new Node
+        if (this.table[index] == null) {
+            Node node = new Node(h, key, value, null);
+            this.table[index] = node;
+        }
+        else{
+            Node oldNode = table[index];
+            if (oldNode.key!=key && oldNode.value!=value){
+                Node temp1 = oldNode;
+                Node temp2 = oldNode.next;
+                if (temp2 == null) temp1.next = new Node(h, key, value, null);
+                else{
+                    while (temp2!=null){
+                        temp1 = temp1.next;
+                        temp2 = temp2.next;
+                    }
+                    temp1.next = new Node(h, key, value, null);
+                }
+            }
+        }
+    }
     @Override
     public V getValue(K key) {
-        return null;
+
+        V result = null;
+
+        for (int i = 0; i < table.length; i++) {
+
+            Node node = table[i];
+
+            // If in this bucket there is only one Node we check it
+            if (node instanceof Node) {
+                if (node.next == null) {
+                    if (node.key instanceof Object && compareObjects(node.key, key)) {
+                        result = (V) node.value;
+                    } else if (node.key == key) {
+                        result = (V) node.value;
+                    }
+                }
+
+                // If in this bucket there is more than one Node we check all of them
+                else {
+                    while (node != null) {
+                        if (node.key instanceof Object && compareObjects(node.key, key)) {
+                            result = (V) node.value;
+                        } else if (node.key == key) {
+                            result = (V) node.value;
+                        }
+                        node = node.next;
+                    }
+                }
+
+            }
+
+        }
+        return result;
```

Comment:
To find an index of a Node you just need to get the remainder of the division key's hashcode by table length, consider implementing a separate ```calculateIndex``` method for this purpose and reuse in other methods